### ISSUE TITLE
Lint/AmbiguousBlockAssociation's message is confusing

### ISSUE THREAD

**Issue by eregon** [rubocop-hq/rubocop#5170]
The message is (for the code `using Module.new {}`):
```
Parenthesize the param Module.new {} to make sure that the block
will be associated with the Module.new method call.
```

But this sounds like spreading fear and doubt when the Ruby parser has no such concepts :)
`{}` always binds tightly to the call right next to it.
This block `{}` will *never* go to using, unless it's rewritten as `do ... end`.

## Expected behavior

A better message, clarifying the true purpose of the cop, which I guess is to differentiate empty Hash literals and blocks.
Maybe
```ruby
core/module/using_spec.rb:92:9: W: Lint/AmbiguousBlockAssociation: Parenthesize the param
`Module.new {}` to clarify `{}` is a block passed to `Module.new` and not an empty Hash literal.
        using Module.new {}
        ^^^^^^^^^^^^^^^^^^^
```

However, I find `using(Module.new {})` not so beautiful, but that might be subjective.
```ruby
mod = Module.new
using mod
```
is likely a better fix.

## Actual behavior

```ruby
core/module/using_spec.rb:92:9: W: Lint/AmbiguousBlockAssociation: Parenthesize the param
Module.new {} to make sure that the block will be associated with the Module.new method call.
        using Module.new {}
        ^^^^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

Run RuboCop on a file containing
```ruby
using Module.new {}
```

## RuboCop version

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.2, running on ruby 2.4.2 x86_64-linux)
```

**Comment by Drenmi** [comment]
Looks like the message was improved once in #4181, but I agree that there's more room for improvement. 🙂

**Comment by bbatsov** [comment]
I'm closing this ticket due to no recent activity. Feel free to re-open it if you ever come back to it. PRs welcome!
